nixos/modules/services/misc/memos.nix

```
@@ -0,0 +1,197 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.memos;
+  envFileFormat = pkgs.formats.keyValue { };
+in
+{
+  options.services.memos = {
+    enable = lib.mkEnableOption "Memos note-taking";
+    package = lib.mkPackageOption pkgs "Memos" {
+      default = "memos";
+    };
+
+    openFirewall = lib.mkEnableOption "opening the ports in the firewall";
+
+    user = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        The user to run Memos as.
+
+        ::: {.note}
+        If changing the default value, **You** are responsible of creating the corresponding user with {option}`users.users`.
+        :::
+      '';
+      default = "memos";
+    };
+
+    group = lib.mkOption {
+      type = lib.types.str;
+      description = ''
+        The group to run Memos as.
+
+        ::: {.note}
+        If changing the default value, **You** are responsible of creating the corresponding group with {option}`users.groups`.
+        :::
+      '';
+      default = "memos";
+    };
+
+    dataDir = lib.mkOption {
+      default = "/var/lib/memos/";
+      type = lib.types.path;
+      description = ''
+        Specifies the directory where Memos will store its data.
+
+        ::: {.note}
+        It will be automatically created with the permissions of {option}`services.memos.user` and {option}`services.memos.group`.
+        :::
+      '';
+    };
+
+    settings = lib.mkOption {
+      type = envFileFormat.type;
+      description = ''
+        The environment variables to configure Memos.
+
+        ::: {.note}
+        At time of writing, there is no clear documentation about possible values.
+        It's possible to convert CLI flags into these variables.
+        Example : CLI flag "unix-sock" converts to {env}`MEMOS_UNIX_SOCK`.
+        :::
+      '';
+      example = {
+        MEMOS_MODE = "prod";
+        MEMOS_ADDR = "127.0.0.1";
+        MEMOS_PORT = "5230";
+        MEMOS_DATA = "/var/lib/memos/";
+        MEMOS_DRIVER = "sqlite";
+        MEMOS_INSTANCE_URL = "http://localhost:5230";
+      };
+      default = { };
+    };
+
+    environmentFile = lib.mkOption {
+      type = lib.types.path;
+      description = ''
+        The environment file to use when starting Memos.
+
+        ::: {.note}
+        By default, generated from {option}`services.memos.settings`.
+        :::
```

No objection to having this note in the description. However it feels a _little_ redundant to repeat the `default` text (which is shown immediately after the description).

If you do have it in both places, this may be a good spot to have a clickable link:

```suggestion
        ::: {.note}
        By default, generated from [](opt-services.memos.settings).
        :::
```

or

```suggestion
        ::: {.note}
        By default, generated from [](opt-${opt.settings}).
        :::
```

---

> I kept the second suggested change, and removed the `defaultText`

Sorry for the confusion:

If the option declares a non-static `default`, then we **must** also declare `defaultText`. If we don't declare `defaultText`, the docs will evaluate `default` and render it as the documented default value.

There are two solutions here:

- Don't pass `default` to `mkOption` and instead define an undocumented default via a `mkOptionDefault` config definition.
- Pass both `default` and `defaultText` to `mkOption`

It's fine to have an undocumented default if the `description` makes the default clear. Although, my personal preference is _usually_ to use the standard `default` and `defaultText` attributes for defining and documenting the default.

---

Yup, I indeed removed the defaultText and CI wasn't happy about it x) So i used your suggestion of 

```nix
defaultText = lib.literalMD ''
  generated from {option}`${opt.settings}`
'';
```